airgun/views/common.py

```diff
@@ -56,21 +56,28 @@ def select_logout(self):
         self.account_menu.click()
         self.logout.click()
 
-    def read(self, widget_names=None):
+    def read(self, widget_names=None, limit=None):
         """Reads the contents of the view and presents them as a dictionary.
 
         :param widget_names: If specified, will read only the widgets names in the list.
+        :param limit: how many entries to fetch at most
 
         :return: A :py:class:`dict` of ``widget_name: widget_read_value``
             where the values are retrieved using the :py:meth:`Widget.read`.
         """
         if widget_names is None:
+            if limit is not None:
+                raise NotImplementedError("You must specify widgets to be able to specify limit")
```

Comment:
s/widgets/widget_names is that right ?

Reply:
Well... you specify widgets by their names